### PR TITLE
WIP channels list

### PR DESCRIPTION
**Pull request by calvinmetcalf**
works by creating a channel called channels, which you send a message to when you join a channel, currently messages received from this get intercepted and turned into messages from the channel in question that simply say *joined* this has the problem of making everyone one join every channel and is likely not what we want. 

Next step is going to be to add a 3rd state of channel knowledge for known about but not joined

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
added, so now channels that you know about, but haven't been in are marked in grey and you only really join them when you click on them, currently when you leave one you only really leave it until you sign back into the app

**Comment by calvinmetcalf** [comment]
still need to make sure only verified users can make channels

File: lib/elements/channels.js [hunk 1]
```diff
@@ -26,8 +26,10 @@ inherits(Channels, BaseElement)
 Channels.prototype.render = function (channels) {
   var self = this
 
-  channels = channels.map(function (channel) {
-    var className = channel.active ? 'active' : ''
+  channels = channels.filter(function (channel) {
+    return channel.name !== 'channels';
```
Comment (knownasilya) [inline lib/elements/channels.js]: semicolon

File: app.js [hunk 2]
```diff
@@ -190,12 +209,23 @@ function App (el) {
...
+        }
         self.data.messages = channel.messages
         self.data.activeChannel = channel
         if (channel.name !== 'friends') db.channels.put(channel.name, {name: channel.name, id: channel.id})
+        return;
       }
+      if (channel.active && !channel.joined) {
+        channel.joined = true;
+      }
+      channel.active = false;
```
Comment (knownasilya) [inline app.js]: semicolons

**Comment by maxogden** [comment]
I think this should be opt-in in the UI, so that channels are not broadcasted by default.

**Comment by calvinmetcalf** [comment]
@maxogden good idea

**Comment by calvinmetcalf** [comment]
closing for now not sure this is the best approach

**Pull request closed without merging**